pyam/core.py

```diff
@@ -510,14 +517,125 @@ def check_aggregate(self, variable, components=None, units=None,
                                        **kwargs)]
 
         if len(diff):
-            msg = '{} of {} data points are not aggregates of components'
-            logger().info(msg.format(len(diff), len(df_variable)))
+            msg = '{} - {} of {} data points are not aggregates of components'
+            logger().info(msg.format(variable, len(diff), len(df_variable)))
 
             if exclude_on_fail:
                 self._exclude_on_fail(diff.index.droplevel([2, 3]))
 
+            diff = pd.concat([diff], keys=[variable], names=['variable'])
```

Comment:
I just wanted to add the variable back onto the data table which is returned. For the wrapping function, you want to be able to see which variables failed (again, happy to move this wrapping to `internal_consistency` to separate the logic)

Reply:
so is this the same as diff.set_index('variable')?

Reply:
In any case, if it's for a specific use case, it is probably best to keep this function the same and adjust where you need it elsewhere.

Reply:
> so is this the same as `diff.set_index('variable')`?

It may well be (I don't know what I'm doing half the time with pandas), I'm just trying to add an extra row to the output index.

> In any case, if it's for a specific use case, it is probably best to keep this function the same and adjust where you need it elsewhere.

Cool so pending danielhuppmann's thoughts a lot will probably get moved into `check_internal_consistency`

Reply:
> so is this the same as `diff.set_index('variable')`?

I think it would be except that `diff` doesn't have a `variable` column at this point so we can't use `set_index` and instead have to do this `pd.concat`.

> In any case, if it's for a specific use case, it is probably best to keep this function the same and adjust where you need it elsewhere.

My opinion is still that it would be better to actually see the thing you're checking consistency for in the output but will leave resolution up to @danielhuppmann